boa_engine/src/syntax/parser/function/mod.rs

```diff
@@ -153,11 +156,24 @@ where
             )));
         }
 
-        Ok(FormalParameterList {
-            parameters: params.into_boxed_slice(),
-            is_simple,
-            has_duplicates,
-        })
+        let mut flags = FormalParameterListFlags::empty();
+        if is_simple {
+            flags |= FormalParameterListFlags::IS_SIMPLE;
+        }
+        if has_duplicates {
+            flags |= FormalParameterListFlags::HAS_DUPLICATES;
+        }
+        if has_rest_parameter {
+            flags |= FormalParameterListFlags::HAS_REST_PARAMETER;
+        }
+        if has_expressions {
+            flags |= FormalParameterListFlags::HAS_EXPRESSIONS;
+        }
+        if has_arguments {
+            flags |= FormalParameterListFlags::HAS_ARGUMENTS;
+        }
```

Comment:
I think this would be more efficient if it's set when the booleans are defined. So, instead of using those booleans, add the flags when they are defined. What do you think?

Reply:
So obvious I didn't even notice it :D